storage/replica_command.go

```diff
@@ -414,9 +414,8 @@ func (r *Replica) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args
 				}
 			} else if ct.ModifiedSpanTrigger != nil {
 				if ct.ModifiedSpanTrigger.GetSystemDBSpan() {
-					// Check if we need to gossip the system config, but do it in a separate
-					// goroutine, and after the batch has been committed.
-					batch.Defer(func() { go r.maybeGossipSystemConfig() })
+					// Check if we need to gossip the system config.
```

Comment:
Why did you get rid of the extra goroutine? It doesn't look like doing it in the current goroutine will be a problem; I'm just wondering why it was in a goroutine before and why it's changing.